src/main/java/org/kuali/kfs/sys/rest/resource/businessobject/BusinessObjectsControllersHelperService.java

```diff
@@ -0,0 +1,423 @@
+            final Control control = attribute.getControl();
+            if (control == null) {
+                continue;
+            }
+
+            final String singleAttributeName = attribute.getName();
+
+            if (control.getType() == Control.Type.TREE) {
+                // we have to do this bean resolution here b/c batch file (the only tree) is still a snowflake
+                // and the DDD doesn't do the bean lookup for us (mainly b/c of the typing); hope to get rid of the
+                // need for special VF type eventually
+                final String valuesFinderName = control.getValuesFinderName();
+                if (StringUtils.isBlank(valuesFinderName)) {
+                    LOG.warn(
+                            "getLookupValues(...) - Cannot find a ValuesFinder : businessObjectName={}; attributename={}",
+                            businessObjectEntry::getName,
+                            () -> singleAttributeName
+                    );
+                    continue;
+                }
+                final HierarchicalControlValuesFinder valuesFinder =
+                        dataDictionaryService.getDDBean(HierarchicalControlValuesFinder.class, valuesFinderName);
+                if (valuesFinder == null) {
+                    LOG.warn(
+                            "getLookupValues(...) - Cannot find a HierarchicalControlValuesFinder : businessObjectName={}; attributename={}",
+                            businessObjectEntry::getName,
+                            () -> singleAttributeName
+                    );
+                    continue;
+                }
+                final List<HierarchicalData> values = valuesFinder.getHierarchicalControlValues();
+                valuesMap.put(singleAttributeName, values);
+            } else {
+                final KeyValuesFinder valuesFinder = control.getValuesFinder();
+                if (valuesFinder == null) {
+                    continue;
+                }
+                // CU Customization: keyValues list now comes from the helper method below.
+                final List<KeyValue> keyValues = getKeyValuesForLookup(valuesFinder);
+                valuesMap.put(singleAttributeName, keyValues);
+            }
+        }
+
+        LOG.debug("getLookupValues(...) - Exit : response={}", valuesMap);
+        return valuesMap;
+    }
+
+    /*
+     * CU Customization: Added this method and the one right below it,
+     * to forcibly add a blank key-value entry if the values finder does not return one.
+     */
+    private List<KeyValue> getKeyValuesForLookup(final KeyValuesFinder valuesFinder) {
+        final List<KeyValue> keyValues = valuesFinder.getKeyValues();
+        if (hasEntryForBlankKey(keyValues)) {
+            return keyValues;
+        } else {
+            final KeyValue blankKeyValue = new ConcreteKeyValue(StringUtils.EMPTY, StringUtils.EMPTY);
+            return Stream.concat(Stream.of(blankKeyValue), keyValues.stream())
```

Comment:
Do we need to do anything to force the empty key value pair to be the first item in the returned list?

Reply:
This particular setup should already return the empty key/value pair as the first one. `Stream.of()` and `Collection.stream()` both return sequential streams, so their concatenation via `Stream.concat()` should also be a sequential stream.